### ISSUE TITLE
NPE with KerasFlatten

### ISSUE THREAD

**Issue by carlos-aguayo** [deeplearning4j/deeplearning4j#3664]
#### Issue Description

I have this Model in Python:

```
model = Sequential()
model.add(Embedding(num_words,
                    EMBEDDING_DIM,
                    weights=[embedding_matrix],
                    input_length=MAX_SEQUENCE_LENGTH,
                    trainable=False))

model.add(Flatten())
model.add(Dense(128, activation='relu'))
model.add(Dense(1, activation='sigmoid'))

model.compile(loss='binary_crossentropy',
              optimizer='adam',
              metrics=['accuracy'])
```

When I import it using DL4J, I get to this part:
https://github.com/deeplearning4j/deeplearning4j/blob/d5bb984b19e9e3125f5d8d76999627f021084083/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/layers/KerasFlatten.java#L68

The "InputType" is of type "InputTypeFeedForward" and thus the method returns null.
https://deeplearning4j.org/doc/org/deeplearning4j/nn/conf/inputs/InputType.InputTypeFeedForward.html
Causing an exception in "getOutputType"
https://github.com/deeplearning4j/deeplearning4j/blob/d5bb984b19e9e3125f5d8d76999627f021084083/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/layers/KerasFlatten.java#L101

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 0.8.0-2

**Comment by saudet** [comment]
@turambar Is this a bug or just something that wasn't implemented because it wouldn't work straightforwardly anyway?

**Comment by maxpumperla** [comment]
@carlos-aguayo in DL4J the output type of an embedding layer is already of FeedForward type, hence the KerasFlatten preprocessor in its current form is not suited for the job (it returns `null`, because it expects to only flatten RNN and CNN types).

I haven't found a solution to this yet, but just wanted to let you know that I keep investigating. Fwiw, if you drop the flatten in the above model, it works as intended.

**Comment by carlos-aguayo** [comment]
Thanks for the update!

> Fwiw, if you drop the flatten in the above model, it works as intended.
I'll try but I believe that the Embedding returns a 2x2 matrix and that you need the Flatten to make it 1D in order to feed it to the neural network in Dense(128

**Comment by maxpumperla** [comment]
@carlos-aguayo I'm reasonably sure this is closed with #4936 (sorry, this slipped my attention)

**Comment by carlos-aguayo** [comment]
sounds good, thanks!
